src/chocolatey/infrastructure.app/nuget/NugetList.cs

```diff
@@ -48,12 +64,11 @@ public static IEnumerable<IPackage> GetPackages(ChocolateyConfiguration configur
                 results = results.Skip(configuration.ListCommand.PageSize * configuration.ListCommand.Page.Value).Take(configuration.ListCommand.PageSize);
             }
 
-            return results.OrderBy(p => p.Id)
-                        .AsEnumerable()
-                        .Where(PackageExtensions.IsListed)
-                        .Where(p => configuration.Prerelease || p.IsReleaseVersion())

```

Comment:
> Earlier when @Jaykul was working on some of this that was similar I could not get the results to match. It wasn't necessarily the count, although it was off as well. It was the order coming back from the query, it was off. #224 (comment) and #224 (comment)

Reviewing them, I'm going to run into the exact same issue, but I'm also unwilling to add distinct_last back. I'd much rather find a way to detect whether if the underlying provider is sourcing from a server than make Search a blocking operation.